compiler/fir/resolve/testData/resolve/smartcasts/bangbang.txt

```diff
@@ -4,28 +4,12 @@ FILE: bangbang.kt
 
     }
     public final fun test_0(a: R|A?|): R|kotlin/Unit| {
-        when (lval <bangbang>: R|A?| = R|<local>/a|) {
-            ==($subj$, Null(null)) ->  {
-                throw R|kotlin/KotlinNullPointerException.KotlinNullPointerException|()
-            }
-            else ->  {
-                R|<local>/<bangbang>|
-            }
-        }
-        .R|/A.foo|()
+        !!(R|<local>/a|).R|/A.foo|()
```

Comment:
Add test case with bangbang over type parameter to this test please

```
fun <X : A?> test_6(x: X) {
    x!!.foo()
}
```

Reply:
This was already covered in some other test (`compiler/testData/ir/irText/expressions/bangbang.kt`) but anyway I will add it here too.

Reply:
I guess it's not quite the same, so I am adding both tests here.

Reply:
It's good to have such testcase in diagnostics tests because they are they run much more often than fir2ir tests

Reply:
Please ignore what I said as they don't make sense; the test in the other file does not call a function and is just using `Any`.